workflow/controller/operator.go

```diff
@@ -276,6 +266,19 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 		woc.computeMetrics(woc.execWf.Spec.Metrics.Prometheus, localScope, realTimeScope, true)
 	}
 
+	// Reconciliation of Outputs (Artifacts). See reportOutputs() of executor.go
+	woc.taskResultReconciliation()
+
+	// check to see if we can do garbage collection of Artifacts;
+	if err := woc.garbageCollectArtifacts(ctx); err != nil {
+		woc.log.WithError(err).Error("failed to GC artifacts")
+		return
+	}
+
+	if woc.wf.Labels[common.LabelKeyCompleted] == "true" { // abort now, we do not want to perform any more processing on a complete workflow because we could corrupt it
+		return
+	}
+
```

Comment:
doing `taskResultReconciliation()` right before garbage collecting seems like it should have an overall effect of speeding up the process of garbage collection

Reply:
Yeah, since garbage collection should execute on the first pass through the operator loop.